cirq-core/cirq/transformers/routing/initial_mapper.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from typing import TYPE_CHECKING, Dict
+import abc
+
+from cirq.value import value_equality
+
+if TYPE_CHECKING:
+    import cirq
+
+
+@value_equality
+class AbstractInitialMapper(metaclass=abc.ABCMeta):
+    """Base class for creating custom initial mapping strategies."""
+
+    @abc.abstractmethod
+    def initial_mapping(self) -> Dict['cirq.Qid', 'cirq.Qid']:
+        """Maps the logical qubits of a circuit onto physical qubits on a device.
+        Returns:
+          qubit_map: the initial mapping of logical qubits to physical qubits.
+        """
+
+    def __str__(self) -> str:
+        return f'Logical to physical qubits: {self.initial_mapping()}'
+
+    def _value_equality_values_(self):
+        return self.initial_mapping()
+
+
+class IdentityInitialMapper(AbstractInitialMapper):
```

Comment:
IdentityMapper strongly implies that this will do a mapping of the form `q -> q`. I was surprised to see it actually do `q -> map[q]`. 

There's a `HardcodedQubitMapper` in cirq-google: https://github.com/quantumlib/Cirq/blob/master/cirq-google/cirq_google/workflow/qubit_placement.py#L111

The semantics are slightly different -- namely it requires a NamedTopology.

Reply:
I don't think we will need a function that does `q -> q` in the future (but we might need one that does `q -> map[q]` given some hardcoded map, for SABRE algorithm for example.). So I've changed the name of the function to `HardCodedInitialMapper` if that's okay.

Reply:
sounds good